core/src/main/java/org/apache/druid/timeline/DataSegment.java

```diff
@@ -100,7 +100,7 @@
   /**
    * Stores some configurations of the compaction task which created this segment.
    * This field is filled in the metadata store only when "storeCompactionState" is set true in the context of the
-   * compaction task which is false by default.
+   * compaction task (true by default).
```

Comment:
super nit - to link to where the default is.
```suggestion
   * compaction task (true by default). {@link org.apache.druid.indexing.common.task.Tasks#DEFAULT_STORE_COMPACTION_STATE}.
```

Reply:
Done